Extension/src/SSH/commandInteractors.ts

```diff
@@ -326,6 +331,25 @@ export class ContinueOnInteractor implements IInteractor {
     }
 }
 
+export class ConnectionFailureInteractor implements IInteractor {
+    static ID = 'connectionFailure';
+
+    constructor(private readonly hostName: string) { }
+
+    get id(): string {
+        return ConnectionFailureInteractor.ID;
+    }
+
+    async onData(data: string): Promise<IInteraction> {
+        const result: IInteraction = { postAction: 'keep' };
+        if (data.includes('Connection refused') || data.includes('Could not resolve hostname')) {
```

Comment:
Is this translated into other languages?  Is there a better way to know it failed (e.g. error code) and message that to the user?

Reply:
The SSH client normally only has English. Other checks are parsing the output as well.

I'm not aware of error codes, and checking exit code doesn't work - even scp and ssh don't agree on the exit code.

Reply:
ok